FirebaseVertexAI/Sources/Types/Internal/APIConfig.swift

```diff
@@ -0,0 +1,87 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Configuration for the generative AI backend API used by this SDK.
+struct APIConfig: Sendable, Hashable {
+  /// The service to use for generative AI.
+  ///
+  /// This controls which backend API is used by the SDK.
+  let service: Service
+
+  /// The specific network address to use for API requests.
+  ///
+  /// This must correspond with the API set in `service`.
+  let serviceEndpoint: ServiceEndpoint
```

Comment:
should `serviceEndpoint` be a member of `service`?

Reply:
e.g. 
```swift
  enum Service {
    case vertexAI(endpoint: VertexAIEndpoint)

    case vertexAI(endpoint: DeveloperEndpoint)
```

Reply:
I refactored this into:
```
enum Service {
  case vertexAI
  case developer(endpoint: Endpoint)
}
```
At least for now, the endpoint is hardcoded to `.firebaseVertexAIProd` for `.vertexAI`. For `.developer` we will need to test on all 3 `Endpoint` cases in the coming weeks so I didn't split it off into a separate type.